### PR TITLE
[docs] DocSearch with autocomplete V1 - POC tree results

### PR DESCRIPTION
**Pull request by Shipow**
image

**Summary**
An experimentation to rebuild DocSearch with Autocomplete V1 in alpha stage.

**Result**
Support of JSX template is game changer, the lib meet our expectations so far.
I believe the className mapping could be better if it could accept a function to define the items classes dynamically (haven't tried the source approach yet).

ps: Code is super weak, basically needs a refacto if the experience convinces.

### CODE REVIEW

**Comment by francoischalifour** [comment]
> I believe the className mapping could be better if it could accept a function to define the items classes dynamically (haven't tried the source approach yet).

@Shipow Do you mind opening an issue on the [forked repo](https://github.com/francoischalifour/autocomplete.js/tree/next) to explain what you mean?

File: docs/styling.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ DocSearch default colorscheme comes in a grey theme with blue highlight.
 
 ## Colorscheme
 
-<img src="https://docsearch.algolia.com/img/assets/default-colorscheme.png" alt="Default colorscheme"/>
+<img src="https://docsearch.algolia.com/img/assets/default-colorscheme.png" width="550px" alt="Default colorscheme"/>
```
Comment (s-pace) [inline docs/styling.md:9]:
```suggestion
<img src="https://docsearch.algolia.com/img/assets/default-colorscheme.png" alt="Default colorscheme"/>
```
Let's avoid to have raw size within the code. I do not see any benefit of setting this size from a .md file.

**Review by s-pace — CHANGES_REQUESTED** [review]

File: docs/styling.md (hunk 1 above)
Comment (Shipow) [reply docs/styling.md:9]: what do you mean ?

Comment (endiliey) [reply docs/styling.md:9]: `width="550px"`

Comment (Shipow) [reply docs/styling.md:9]: I'm curious about the "Let's avoid to have raw size".
For instance this image has a retina resolution, on a regular screen it's displayed too big and blurry. We should probably fix that with a better retina support but i'm curious about this practice of not having any width for images in .md

Comment (francoischalifour) [reply docs/styling.md:9]: I guess @s-pace's point was to either resize the source image or use CSS styling.

Comment (Shipow) [reply docs/styling.md:9]: oh I see, github markdown engine doesn't handle other attribute than alt. ok
makes sens now :)

Comment (s-pace) [reply docs/styling.md:9]: Exactly. Having a hardcoded width here will not solve the retina support. I do agree we would need to enhance this support. Another point is that one can view this .md from GitHub. We do not want to force the width.
image

Comment (s-pace) [reply docs/styling.md:9]: Alright point solved after IRL discussion. The image was captured on a retina screen with a 2X resolution. Setting `width` attribute of an image is a good practice and this value is the half of the original image width.

**Review by s-pace — APPROVED** [review]

**Review by s-pace — APPROVED** [review]
LGTM

**Pull request merged**